core/font.lua

```diff
@@ -4,10 +4,58 @@ local icu = require("justenoughicu")
 
 local lastshaper
 
+local bits = require("core.parserbits")
+local lpeg = require("lpeg")
+local Ct, Cg, P = lpeg.Ct, lpeg.Cg, lpeg.P
+local adjust_metric = P("ex-height") + P("cap-height")
+local adjustment = Ct(Cg(bits.number, "amount")^-1 * bits.ws * Cg(adjust_metric, "unit"))
+
+local function measureFontAdjustment (metric)
+   if metric == "ex-height" then
+      -- Uses the height of lowercase letters.
+      -- This is used to normalize lowercase letters across fonts.
+      -- The height of the lowercase letter "x" is used as the reference.
+      -- Another option would be to use the OS/2 font table sxHeight value when available.
+      return SILE.shaper:measureChar("x").height
+   end
+   if metric == "cap-height" then
+      -- Uses the the height of uppercase letters.
+      -- This is used to normalize uppercase letters across fonts.
+      -- The height of the uppercase letter "H" is used as the reference.
+      -- Another option would be to use the OS/2 font table sCapHeight value when available.
+      return SILE.shaper:measureChar("H").height
+   end
+   SU.error("Unknown font adjust metric " .. metric)
+end
+
+local function adjustedFontSize(options)
+   local adjust = options.adjust
+   local parsed = adjustment:match(adjust)
+   if not parsed then
+      SU.error("Couldn't parse font adjust value " .. adjust)
+   end
+   local baseOpts = pl.tablex.copy(options) -- shallow copy
```

Comment:
If need be, add a comment for the rationale for the shallow copy vs. in-place modification:
We shouldn't alter the original AST, as these things may end-up been propagated elsewhere (e.g. to running headers etc.).

Reply:
Comment added.